.github/workflows/knative-downstream.yaml

```diff
@@ -80,19 +80,34 @@ jobs:
       uses: actions/setup-go@v2
       with:
         go-version: 1.16.x
+
     - name: Install Dependencies
       run: |
         go get github.com/google/go-licenses
-    - name: Checkout Upstream
+
+    - name: Checkout upstream at ${{ github.base_ref }}
       uses: actions/checkout@v2
       with:
+        fetch-depth: 0
         path: ./src/knative.dev/${{ github.event.repository.name }}
-    - name: Checkout Downstream
+
+    - name: Merge upstream
```

Comment:
Why is using ref in the checkout not enough?

Reply:
after a PR sits for a few days we want to retest with any changes `HEAD`

Reply:
especially if there are failed checks - this what prow does as well